src/components/mgt-teams-channel-picker/mgt-teams-channel-picker.ts

```diff
@@ -0,0 +1,794 @@
+    if (this._selectedItemState) {
+      return { channel: this._selectedItemState.item, team: this._selectedItemState.parent.item };
+    } else {
+      return null;
+    }
+  }
+
+  private set items(value) {
+    if (this._items === value) {
+      return;
+    }
+    this._items = value;
+    this._treeViewState = value ? this.generateTreeViewState(value) : [];
+    this.resetFocusState();
+  }
+  private get items(): DropdownItem[] {
+    return this._items;
+  }
+
+  // User input in search
+  @query('.team-chosen-input') private _input: HTMLInputElement;
+  private _inputValue: string = '';
+
+  private _isFocused = false;
+
+  private _selectedItemState: ChannelPickerItemState;
+  private _items: DropdownItem[];
+  private _treeViewState: ChannelPickerItemState[] = [];
+
+  // focus state
+  private _focusList: ChannelPickerItemState[] = [];
+  private _focusedIndex: number = -1;
+  private debouncedSearch;
+
+  // determines loading state
+  @property({ attribute: false }) private _isDropdownVisible;
+
+  constructor() {
+    super();
+    this.handleWindowClick = this.handleWindowClick.bind(this);
+    this.addEventListener('keydown', e => this.onUserKeyDown(e));
+    this.addEventListener('focus', _ => this.loadTeamsIfNotLoaded());
+    this.addEventListener('mouseover', _ => this.loadTeamsIfNotLoaded());
+  }
+
+  /**
+   * Invoked each time the custom element is appended into a document-connected element
+   *
+   * @memberof MgtTeamsChannelPicker
+   */
+  public connectedCallback() {
+    super.connectedCallback();
+    window.addEventListener('click', this.handleWindowClick);
+  }
+
+  /**
+   * Invoked each time the custom element is disconnected from the document's DOM
+   *
+   * @memberof MgtTeamsChannelPicker
+   */
+  public disconnectedCallback() {
+    window.removeEventListener('click', this.handleWindowClick);
+    super.disconnectedCallback();
+  }
+
+  /**
+   * selects a channel by looking up the id in the Graph
+   *
+   * @param {string} channelId MicrosoftGraph.Channel.id
+   * @returns {Promise<return>} A promise that will resolve to true if channel was selected
+   * @memberof MgtTeamsChannelPicker
+   */
+  public async selectChannelById(channelId: string): Promise<boolean> {
+    const provider = Providers.globalProvider;
+    if (provider && provider.state === ProviderState.SignedIn) {
+      // since the component normally handles loading on hover, forces the load for items
+      if (!this.items) {
+        this.requestStateUpdate();
+      }
+
+      let foundChannel: ChannelPickerItemState;
+
+      compareItem: for (const item of this._treeViewState) {
+        compareChannel: for (const channel of item.channels) {
+          if (channel.item.id !== channelId) {
+            continue compareChannel;
+          } else {
+            foundChannel = channel;
+          }
+          continue compareItem;
+        }
+      }
+      if (foundChannel) {
+        this.selectChannel(foundChannel);
```

Comment:
I like the label additions, but I want to double check that this is doing what you expect it to. When a channel is found, do you want to continue iterating and potentially override it if another channel is found? Or do you just want to find the first viable channel and use it? It looks like you are doing the former, which is different from what the first solution did.

Tbh, I don't actually think you need the labels OR break logic. Try this instead:
```
for (const item of this._treeViewState) {
  for (const channel of item.channels) {
    if (channel.item.id === channelId) {
      this.selectedChannel(channel);
      return true;
    }
  }
}
```